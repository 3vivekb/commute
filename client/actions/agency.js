import fetchAction from '@conveyal/woonerf/fetch'
import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addLocally = createAction('add agency')
export const createAgency = (newAgency) => {
  newAgency.id = uuid.v4()
  newAgency.organizations = []
  return [
    addLocally(newAgency),
    push(`/agency/${newAgency.id}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete agency')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/agency/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteAgency = (id) => [deleteLocally(id), push('/')]

export const loadAgencies = () => fetchAction({
  url: '/api/agency',
  next: (res, err) => {
    if (!err) {
      return setAllLocally(res)
    }
  }
})

export const updateAgency = (agency) => [
  setLocally(agency),
  push(`/agency/${agency.id}`)
]
/* const updateOnServer = (agency) =>
  serverAction({
    url: `/api/agency/${agency.id}`,
    options: {
      body: JSON.stringify(agency),
      method: 'put'
    }
  }) */  // TODO: update on server

export const setLocally = createAction('set agency')
export const setAllLocally = createAction('set agencies')
