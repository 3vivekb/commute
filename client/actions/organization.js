import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addOrganization = createAction('add organization')
export const createOrganization = (newOrganization) => {
  newOrganization.id = uuid.v4()
  newOrganization.sites = []
  newOrganization.groups = []
  return [
    addOrganization(newOrganization),
    push('/')
  ] // TODO save to server
}

// site stuff
const addSite = createAction('add site')
export const createSite = (newSite) => {
  newSite.id = uuid.v4()
  return [
    addSite(newSite),
    push(`/organizations/${newSite.organizationId}`)
  ] // TODO save to server
}
