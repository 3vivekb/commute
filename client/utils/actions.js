import fetchAction from '@conveyal/woonerf/fetch'
import qs from 'qs'
import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'

/**
 * Make generic model actions that communicate with a rest server
 *
 * @param  {Object} cfg An object with the following parameters:
 * - commands: An object with the keys being commands and the corresponding cfg of each command
 * -- redirectionStrategy: The redirection strategy to use upon success.
 *  Can be one of following: 'toEntity', 'toParent', 'toHome', undefined (no redirection)
 *  Defaults are as follows:
 *    Collection GET, GET: null
 *    Collection POST, PUT: toEntity
 *    DELETE: toParent
 * - parentKey: Key in model of parent's id
 *   (must be provided if using toParent redirection strategy)
 * - parentName: Name of app route to parent view
 *   (must be provided if using toParent redirection strategy)
 * - pluralName: Plural name of the model
 * - singularName: Singular name of the model
 * @return {Object}     The actions
 */
const makeGenericModelActions = (cfg) => {
  const {commands, parentKey, parentName, pluralName, singularName} = cfg
  const actions = {}

  // make local set actions
  const addLocally = createAction(`add ${singularName}`)
  const deleteLocally = createAction(`delete ${singularName}`)
  const setLocally = createAction(`set ${singularName}`)
  const setAllLocally = createAction(`set ${pluralName}`)

  const baseEndpoint = `/api/${singularName}`

  const redirectionStrategies = {
    'toEntity': (entity) => {
      return push(`/${singularName}/${entity._id}`)
    },
    'toHome': () => {
      return push('/')
    },
    'toParent': (entity) => {
      return push(`/${parentName}/${entity[parentKey]}`)
    }
  }

  /**
   * Do a redirect if needed
   *
   * @param  {Object} cfg Paremters as follows:
   *   - actions: The actions to append to
   *   - endpointCfg: The endpoint config
   *   - defaultStrategy: The default redire strategy to use (default: undefined)
   *   - redirectArgs: Arguments to provide to the redirectionStrategy
   */
  const doRedirectIfNecessary = (cfg) => {
    const redirectionStrategy = (cfg.endpointCfg.redirectionStrategy !== undefined
      ? cfg.endpointCfg.redirectionStrategy
      : cfg.defaultStrategy)

    if (redirectionStrategy) {
      const redirect = redirectionStrategies[redirectionStrategy](cfg.redirectArgs)
      if (redirect) {
        cfg.actions.push(redirect)
      }
    }
  }

  if (commands['Collection GET']) {
    actions.loadAll = (queryParams) => {
      // only include filteredKeys in querystring
      let queryString = ''
      if (queryParams) {
        queryString = qs.stringify(queryParams)
        if (queryString.length !== 0) {
          queryString = '?' + queryString
        }
      }

      // make request
      return fetchAction({
        next: (res, err) => {
          if (!err) {
            return setAllLocally(res.value)
          }
          // TODO handle error
        },
        url: baseEndpoint + queryString
      })
    }
  }

  if (commands['Collection POST']) {
    const endpointCfg = commands['Collection POST']
    actions.create = (newEntity) => {
      return fetchAction({
        next: (res, err) => {
          if (err) {
            // TODO handle error
          } else {
            const createdEntity = res.value
            const actions = [
              addLocally(createdEntity)
            ]
            doRedirectIfNecessary({
              actions,
              endpointCfg,
              defaultStrategy: 'toEntity',
              redirectArgs: createdEntity
            })
            return actions
          }
        },
        options: {
          body: newEntity,
          method: 'POST'
        },
        url: baseEndpoint
      })
    }
  }

  if (commands['DELETE']) {
    const endpointCfg = commands['DELETE']
    actions.delete = (entity) => {
      return fetchAction({
        next: (res, err) => {
          if (err) {
            // TODO handle error
          } else {
            const actions = [
              deleteLocally(entity._id)
            ]
            doRedirectIfNecessary({
              actions,
              endpointCfg,
              defaultStrategy: 'toParent',
              redirectArgs: entity
            })
            return actions
          }
        },
        options: {
          method: 'DELETE'
        },
        url: `${baseEndpoint}/${entity._id}`
      })
    }
  }

  if (commands['GET']) {
    actions.loadOne = (id) => fetchAction({
      next: (res, err) => {
        if (!err) {
          return setLocally(res.value)
        }
        // TODO handle error
      },
      url: `${baseEndpoint}/${id}`
    })
  }

  if (commands['PUT']) {
    const endpointCfg = commands['PUT']
    actions.update = (entity) => fetchAction({
      next: (res, err) => {
        if (err) {
          // TODO handle error
        } else {
          const updatedEntity = res.value
          const actions = [
            setLocally(updatedEntity)
          ]
          doRedirectIfNecessary({
            actions,
            endpointCfg,
            defaultStrategy: 'toEntity',
            redirectArgs: updatedEntity
          })
          return actions
        }
      },
      options: {
        body: entity,
        method: 'PUT'
      },
      url: `${baseEndpoint}/${entity._id}`
    })
  }

  return actions
}

export default makeGenericModelActions
