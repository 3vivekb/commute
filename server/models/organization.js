const {Schema} = require('mongoose')

const Organization = module.exports = new Schema({
  createdBy: String,
  name: String
})

Organization.methods.update = function (properties) {
  Object
    .keys(properties)
    .forEach((key) => {
      this[key] = properties[key]
    })
  return this.save()
}
