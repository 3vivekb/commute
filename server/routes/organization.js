const models = require('../models')
const makeRestEndpoints = require('../utils/restEndpoints')

module.exports = function makeRoutes (app) {
  makeRestEndpoints(app,
    {
      childModels: [{
        foreignKey: 'organizationId',
        key: 'analyses',
        model: models.Analysis
      }, {
        foreignKey: 'organizationId',
        key: 'groups',
        model: models.Group
      }, {
        foreignKey: 'organizationId',
        key: 'sites',
        model: models.Site
      }],
      commands: {
        'Collection GET': {},
        'Collection POST': {},
        'GET': {},
        'DELETE': {},
        'PUT': {}
      },
      model: models.Organization,
      name: 'organization'
    }
  )
}
