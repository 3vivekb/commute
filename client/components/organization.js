import React, {Component, PropTypes} from 'react'
import {Button, ButtonGroup, Col, Grid, Row} from 'react-bootstrap'
import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table'
import {Link} from 'react-router'

import Icon from './icon'
import {messages} from '../utils/env'
import {arrayCountRenderer} from '../utils/table'
import {actUponConfirmation} from '../utils/ui'

export default class Organization extends Component {
  static propTypes = {
    // dispatch
    deleteAnalysis: PropTypes.func,
    deleteGroup: PropTypes.func,
    deleteOrganization: PropTypes.func,
    deleteSite: PropTypes.func,

    // props
    analysis: PropTypes.object.isRequired,
    analyses: PropTypes.array.isRequired,
    group: PropTypes.object.isRequired,
    groups: PropTypes.array.isRequired,
    organization: PropTypes.object.isRequired,
    site: PropTypes.object.isRequired,
    sites: PropTypes.array.isRequired
  }

  _analysisGroupNameRenderer = (cell, row) => {
    const group = this.props.group[row.groupId]
    return <Link to={`/group/${group.id}`}>{group.name}</Link>
  }

  _analysisNameRenderer = (cell, row) => {
    return <Link to={`/analysis/${row.id}`}>{row.name}</Link>
  }

  _analysisSiteNameRenderer = (cell, row) => {
    const site = this.props.site[row.siteId]
    return <Link to={`/group/${site.id}`}>{site.name}</Link>
  }

  _analysisToolsRenderer = (cell, row) => {
    const doDelete = () => {
      this.props.deleteAnalysis({ id: row.id, organizationId: this.props.organization.id })
    }
    const onClick = () => actUponConfirmation(messages.analysis.deleteConfirmation, doDelete)
    return <Button bsStyle='danger' onClick={onClick}>Delete</Button>
  }

  _handleDelete = () => {
    const doDelete = () => this.props.deleteOrganization(this.props.organization)
    actUponConfirmation(messages.organization.deleteConfirmation, doDelete)
  }

  _groupNameRenderer = (cell, row) => {
    return <Link to={`/group/${row.id}/`}>{row.name}</Link>
  }

  _groupToolsRenderer = (cell, row) => {
    const doDelete = () => this.props.deleteGroup({ id: row.id, organizationId: this.props.organization.id })
    const onClick = () => actUponConfirmation(messages.group.deleteConfirmation, doDelete)
    return <Button bsStyle='danger' onClick={onClick}>Delete</Button>
  }

  _siteNameRenderer = (cell, row) => {
    return <Link to={`/site/${row.id}/edit`}>{row.name}</Link>
  }

  _siteToolsRenderer = (cell, row) => {
    const organizationId = this.props.organization.id
    const doDelete = () => this.props.deleteSite({ id: row.id, organizationId })
    const onClick = () => actUponConfirmation(messages.site.deleteConfirmation, doDelete)
    return <div>
      <Button bsStyle='warning'>
        <Link to={`/site/${row.id}/edit`}>Edit</Link>
      </Button>
      <Button bsStyle='danger' onClick={onClick}>Delete</Button>
    </div>
  }

  render () {
    const {analyses, groups, sites} = this.props
    const {id: organizationId, name} = this.props.organization

    return (
      <Grid>
        <Row>
          <Col xs={12}>
            <h2><Icon type='shield' />{name}</h2>
            <ButtonGroup>
              <Button bsStyle='warning'>
                <Link to={`/organization/${organizationId}/edit`}>Edit</Link>
              </Button>
              <Button bsStyle='danger' onClick={this._handleDelete}>Delete</Button>
            </ButtonGroup>
            <p>Below are this organization's sites, commuter groups and analyses.</p>
            <h3>Sites
              <Button className='pull-right'>
                <Link to={`/organization/${organizationId}/site/create`}>Create a new site <Icon type='building' /></Link>
              </Button>
            </h3>
            <p>A site is a location of a building or new address that you want to use as the centerpoint of your commutes.</p>
            <BootstrapTable data={sites}>
              <TableHeaderColumn dataField='id' isKey hidden />
              <TableHeaderColumn dataFormat={this._siteNameRenderer}>Name</TableHeaderColumn>
              <TableHeaderColumn dataField='address'>Address</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._siteToolsRenderer}>Tools</TableHeaderColumn>
            </BootstrapTable>
            <h3>Commuter Groups
              <Button className='pull-right'>
                <Link to={`/organization/${organizationId}/group/create`}>Create a new group <Icon type='users' /></Link>
              </Button>
            </h3>
            <p>A commuter group is a list of commuters that can commute to a particular site.</p>
            <BootstrapTable data={groups}>
              <TableHeaderColumn dataField='id' isKey hidden />
              <TableHeaderColumn dataFormat={this._groupNameRenderer}>Name</TableHeaderColumn>
              <TableHeaderColumn dataField='commuters' dataFormat={arrayCountRenderer}>Commuters</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._groupToolsRenderer}>Tools</TableHeaderColumn>
            </BootstrapTable>
            <h3>Analyses
              <Button className='pull-right'>
                <Link to={`/organization/${organizationId}/analysis/create`}>Create a new analysis <Icon type='bar-chart' /></Link>
              </Button>
            </h3>
            <p>An analysis calculates commuting statistics for a pairing of a commuter group and site.</p>
            <BootstrapTable data={analyses}>
              <TableHeaderColumn dataField='id' isKey hidden />
              <TableHeaderColumn dataFormat={this._analysisNameRenderer}>Name</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._analysisSiteNameRenderer}>Site</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._analysisGroupNameRenderer}>Group</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._analysisToolsRenderer}>Tools</TableHeaderColumn>
            </BootstrapTable>
          </Col>
        </Row>
      </Grid>
    )
  }
}
