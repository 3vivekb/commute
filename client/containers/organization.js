import {connect} from 'react-redux'

import analysisActions from '../actions/analysis'
import groupActions from '../actions/group'
import organizationActions from '../actions/organization'
import siteActions from '../actions/site'
import Organization from '../components/organization'
import {entityIdArrayToEntityArray} from '../utils/entities'

function mapStateToProps (state, props) {
  const {analysis, group, organization, site} = state
  const currentOrganization = organization[props.params.organizationId]
  return {
    analysis,
    analyses: entityIdArrayToEntityArray(currentOrganization.analyses, analysis),
    group,
    groups: entityIdArrayToEntityArray(currentOrganization.groups, group),
    organization: currentOrganization,
    site,
    sites: entityIdArrayToEntityArray(currentOrganization.sites, site)
  }
}

function mapDispatchToProps (dispatch) {
  return {
    deleteAnalysis: (opts) => dispatch(analysisActions.delete(opts)),
    deleteGroup: (opts) => dispatch(groupActions.delete(opts)),
    deleteOrganization: (opts) => dispatch(organizationActions.delete(opts)),
    deleteSite: (opts) => dispatch(siteActions.delete(opts)),
    loadAnalyses: (opts) => dispatch(analysisActions.loadMany(opts)),
    loadGroups: (opts) => dispatch(groupActions.loadMany(opts)),
    loadSites: (opts) => dispatch(siteActions.loadMany(opts))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Organization)
