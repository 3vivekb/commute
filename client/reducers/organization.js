import {makeChildrenHandlers, makeGenericReducerHandlers} from '../utils/reducers'

export const reducers = makeGenericReducerHandlers({
  handlers: ['add', 'delete', 'set', 'set all'],
  name: {
    singular: 'organization',
    plural: 'organizations'
  }
})

Object.assign(reducers, makeChildrenHandlers({
  childPluralName: 'analyses',
  childSingularName: 'analysis',
  parentIdField: 'organizationId'
}))

Object.assign(reducers, makeChildrenHandlers({
  childPluralName: 'groups',
  childSingularName: 'group',
  parentIdField: 'organizationId'
}))

Object.assign(reducers, makeChildrenHandlers({
  childPluralName: 'sites',
  childSingularName: 'site',
  parentIdField: 'organizationId'
}))

export const initialState = {}
