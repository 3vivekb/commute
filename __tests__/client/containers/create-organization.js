/* global describe, expect, it */

import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'
import React from 'react'
import {Provider} from 'react-redux'

import {expectCreateAction} from '../actions/organization'
import {makeMockStore, mockStores} from '../../test-utils/mock-store.js'

import CreateOrganization from '../../../client/containers/create-organization'

describe('Container > CreateOrganization', () => {
  it('renders correctly', () => {
    const mockStore = makeMockStore(mockStores.init)

    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <CreateOrganization />
      </Provider>
    )
    expect(mountToJson(tree.find(CreateOrganization))).toMatchSnapshot()
  })

  it('can create an organization', () => {
    const mockStore = makeMockStore(mockStores.init)

    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <CreateOrganization />
      </Provider>
    )

    // give each text field some input
    tree.find('input').map((input) => input.simulate('change', {target: {value: 'My new value'}}))

    // submit form
    tree.find('form').find('button').simulate('click')

    // expect create action
    expectCreateAction(mockStore.getActions())
  })
})
