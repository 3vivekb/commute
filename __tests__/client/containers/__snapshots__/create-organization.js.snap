exports[`Container > CreateOrganization can create an organization 1`] = `
Object {
  "payload": Object {
    "contact": "My new value",
    "email": "My new value",
    "groups": Array [],
    "logo_url": "My new value",
    "main_url": "My new value",
    "name": "My new value",
    "sites": Array [],
  },
  "type": "add organization",
}
`;

exports[`Container > CreateOrganization can create an organization 2`] = `
Object {
  "payload": Object {
    "args": Array [
      "/",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`Container > CreateOrganization renders correctly 1`] = `
<Connect(CreateOrganization)>
  <CreateOrganization
    create={[Function]}>
    <Grid
      bsClass="container"
      componentClass="div"
      fluid={false}>
      <div
        className="container">
        <Row
          bsClass="row"
          componentClass="div">
          <div
            className="row">
            <Col
              bsClass="col"
              componentClass="div"
              xs={12}>
              <div
                className="col-xs-12">
                <h3>
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    className="pull-right"
                    disabled={false}>
                    <button
                      className="pull-right btn btn-default"
                      disabled={false}
                      type="button">
                      <Link
                        onlyActiveOnIndex={false}
                        style={Object {}}
                        to="/">
                        <a
                          onClick={[Function]}
                          style={Object {}}>
                          <pure(Icon)
                            type="arrow-left">
                            <i
                              className="fa fa-arrow-left fa-fw " />
                          </pure(Icon)>
                        </a>
                      </Link>
                    </button>
                  </Button>
                </h3>
                <form>
                  <FieldGroup
                    label="Name"
                    name="name"
                    onChange={[Function]}
                    placeholder="Enter name"
                    type="text">
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-name">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-name">
                            Name
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter name"
                          type="text">
                          <input
                            className="form-control"
                            id="group-item-name"
                            onChange={[Function]}
                            placeholder="Enter name"
                            type="text" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <FieldGroup
                    label="Main URL"
                    name="main_url"
                    onChange={[Function]}
                    placeholder="Enter url"
                    type="text">
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-main_url">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-main_url">
                            Main URL
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter url"
                          type="text">
                          <input
                            className="form-control"
                            id="group-item-main_url"
                            onChange={[Function]}
                            placeholder="Enter url"
                            type="text" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <FieldGroup
                    label="Logo URL"
                    name="logo_url"
                    onChange={[Function]}
                    placeholder="Enter logo url"
                    type="text">
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-logo_url">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-logo_url">
                            Logo URL
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter logo url"
                          type="text">
                          <input
                            className="form-control"
                            id="group-item-logo_url"
                            onChange={[Function]}
                            placeholder="Enter logo url"
                            type="text" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <FieldGroup
                    label="Contact"
                    name="contact"
                    onChange={[Function]}
                    placeholder="Enter contact"
                    type="text">
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-contact">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-contact">
                            Contact
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter contact"
                          type="text">
                          <input
                            className="form-control"
                            id="group-item-contact"
                            onChange={[Function]}
                            placeholder="Enter contact"
                            type="text" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <FieldGroup
                    label="Email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter email"
                    type="text">
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-email">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-email">
                            Email
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter email"
                          type="text">
                          <input
                            className="form-control"
                            id="group-item-email"
                            onChange={[Function]}
                            placeholder="Enter email"
                            type="text" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={false}
                    onClick={[Function]}>
                    <button
                      className="btn btn-default"
                      disabled={false}
                      onClick={[Function]}
                      type="button">
                      Create
                    </button>
                  </Button>
                </form>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Grid>
  </CreateOrganization>
</Connect(CreateOrganization)>
`;

exports[`actions > organization create organization should work 1`] = `
Object {
  "payload": Object {
    "groups": Array [],
    "name": "mockOrg",
    "sites": Array [],
  },
  "type": "add organization",
}
`;

exports[`actions > organization create organization should work 2`] = `
Object {
  "payload": Object {
    "args": Array [
      "/",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`actions > organization create site should work 1`] = `
Object {
  "payload": Object {
    "name": "mockSite",
    "organizationId": "1",
  },
  "type": "add site",
}
`;

exports[`actions > organization create site should work 2`] = `
Object {
  "payload": Object {
    "args": Array [
      "/organizations/1",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;
